tests/integration/synapseclient/models/async/test_dataset_async.py

```diff
@@ -549,3 +549,424 @@ async def test_rename_column(self, project_model: Project) -> None:
         )
         assert new_dataset_instance.columns[new_column_name] is not None
         assert old_column_name not in new_dataset_instance.columns
+
+
+class TestDatasetCollection:
+    @pytest.fixture(autouse=True, scope="function")
+    def init(self, syn: Synapse, schedule_for_cleanup: Callable[..., None]) -> None:
+        self.syn = syn
+        self.schedule_for_cleanup = schedule_for_cleanup
+
+    def create_file_instance(self, schedule_for_cleanup: Callable[..., None]) -> File:
+        filename = utils.make_bogus_uuid_file()
+        schedule_for_cleanup(filename)
+        return File(
+            path=filename,
+            description=DESCRIPTION_FILE,
+            content_type=CONTENT_TYPE,
+        )
+
+    @pytest.fixture(autouse=True, scope="function")
+    def file(self, schedule_for_cleanup: Callable[..., None]) -> File:
+        return self.create_file_instance(schedule_for_cleanup)
+
+    def create_dataset_instance(
+        self, schedule_for_cleanup: Callable[..., None], project_model: Project
+    ) -> Dataset:
+        dataset_name = str(uuid.uuid4())
+        schedule_for_cleanup(dataset_name)
```

Comment:
I don't think this causes anything to happen: https://github.com/Sage-Bionetworks/synapsePythonClient/blob/7512c0e6351b079eed5f952b2a606068ea48088c/tests/integration/conftest.py#L155-L178